### PR TITLE
Xinnify

### PR DESCRIPTION
**Pull request by GitTiago**
I had to put my mad doctor hat to make this happen.

So there are 2 repositories with efforts into Django 3. One it's a two commit ahead fork from one our developers which we have hosted ourselves and another is from the developer we forked our zinnia cms plugin from, this one is named Xinnia.

Xinnia however seems to have been a short lived attempt as the tests are not really passing and had at least two bugs to be addressed. Not only that the first Xinnia commit is a copy not a fork from the Zinnia repository.
These two repositories are now merged, fixed and have the changelog merged, the tests have been updated too.

I have not really introduced tox as I wished but I have run the tests with both Django 2.2.28 and 3.2.16.
Unfortunately they don't run with Django 4.

### CODE REVIEW

File: zinnia/signals.py [hunk 1]
```diff
@@ -29,8 +29,10 @@
 PINGBACK_WF_COUNT_PINGBACKS = 'zinnia.pingback.was_flagged.count_pingbacks'
 TRACKBACK_WF_COUNT_TRACKBACKS = 'zinnia.trackback.was_flagged.count_trackbacks'
 
-pingback_was_posted = Signal(providing_args=['pingback', 'entry'])
-trackback_was_posted = Signal(providing_args=['trackback', 'entry'])
+# providing_args=['pingback', 'entry']
+pingback_was_posted = Signal()
+# providing_args=['trackback', 'entry']
+trackback_was_posted = Signal()
```
Comment (GitTiago) [inline zinnia/signals.py:33]: Apparently it stopped being supported

Django [2.2](https://docs.djangoproject.com/en/2.2/topics/signals/#defining-signals) vs [3.2](https://docs.djangoproject.com/en/3.2/topics/signals/#defining-signals)

File: zinnia/search.py [hunk 2]
```diff
@@ -126,7 +126,7 @@ def union_q(token):
                 WILDCARDS.setResultsName('query')))
 TERM.setParseAction(create_q)
 
-EXPRESSION = operatorPrecedence(TERM, [
```
Comment (GitTiago) [inline zinnia/search.py:129]: https://github.com/pyparsing/pyparsing/blob/master/CHANGES#L849

File: zinnia/views/mixins/templates.py [hunk 3]
```diff
@@ -92,31 +90,30 @@ def get_template_names(self):
...
-            for template_name in template_names:
-                templates.append(
-                    '%s/%s/%s' % (path, year, template_name))
-        if week:
+                              f'zinnia/{template_name}',
+                              f'{path}/{template_name}'])
+        if year is not None:
+            templates.extend(f'{path}/{year}/{template_name}' for template_name in template_names)
+
+        if week is not None:
```
Comment (GitTiago) [inline zinnia/views/mixins/templates.py:98]: week went from string that has '00' to 0 as an int. This allows the code to behave as originally intended

File: CHANGELOG [hunk 4]
```diff
@@ -352,3 +389,4 @@ https://github.com/Fantomas42/django-blog-zinnia/compare/v0.12.3...v0.13
 
 * The previous versions of Zinnia were not packaged, and were destinated for a
   personnal use.
```
Comment (horthbynorthwest) [inline CHANGELOG]: Non-blocking: Typo - but also do we need to address this as we're a business?
```suggestion
  personal use.
```

File: zinnia/signals.py (hunk 1 above)
Comment (horthbynorthwest) [reply zinnia/signals.py:33]: Cut the commented lines.

File: zinnia/tests/implementations/urls/custom_detail_views.py [hunk 5]
```diff
@@ -16,8 +17,7 @@ class CustomModelDetailMixin(object):
     template_name = 'zinnia/entry_custom_list.html'
 
     def get_context_data(self, **kwargs):
-        context = super(CustomModelDetailMixin,
-                        self).get_context_data(**kwargs)
```
Comment (horthbynorthwest) [inline zinnia/tests/implementations/urls/custom_detail_views.py:20]: ~~Is this an older way to use `super()`or is this just them being really specific?~~

Chatted offline this is the old way of using `super()` and specifying where super should be called from (i.e. super of myself)

File: zinnia/tests/test_admin.py [hunk 6]
```diff
@@ -191,14 +191,13 @@ def test_get_sites(self):
     def test_get_short_url(self):
         with self.settings(ROOT_URLCONF=self.poor_urls):
             entry_url = self.entry.get_absolute_url()
-
+        hash_ = base36(self.entry.pk)
```
Comment (horthbynorthwest) [inline zinnia/tests/test_admin.py:194]: I can't see where the original `hash` is defined or imported, am I missing something?

File: zinnia/models_bases/entry.py [hunk 7]
```diff
@@ -92,12 +92,7 @@ def is_actual(self):
         Checks if an entry is within his publication period.
         """
         now = timezone.now()
-        if self.start_publication and now < self.start_publication:
-            return False
-
-        if self.end_publication and now >= self.end_publication:
-            return False
-        return True
+        return (self.start_publication or now) <= now <= (self.end_publication or now)
```
Comment (horthbynorthwest) [inline zinnia/models_bases/entry.py:95]: *chef's kiss*

File: zinnia/tests/test_views.py [hunk 8]
```diff
@@ -139,17 +139,14 @@ class ViewsTestCase(ViewsBaseCase):
 
     @override_settings(USE_TZ=False)
     def test_zinnia_entry_archive_index_no_timezone(self):
-        template_name_today = 'zinnia/archives/%s/entry_archive.html' % \
-                              date.today().strftime('%Y/%m/%d')
-        response = self.check_publishing_context(
-            '/', 2, 3, 'entry_list', 2)
+        template_name_today = f"zinnia/archives/{date.today():%Y/%m/%d}/entry_archive.html"
+
+        response = self.check_publishing_context('/', 2, 3, 'entry_list', 2)
```
Comment (horthbynorthwest) [inline zinnia/tests/test_views.py:144]: I know you haven't really changed anything here, but boy my head hurts with this

**Review by horthbynorthwest — COMMENTED** [review]
Will update after practical testing but so far, so good. Long live the zombie! 🧟

File: zinnia/tests/test_admin.py (hunk 6 above)
Comment (GitTiago) [reply zinnia/tests/test_admin.py:194]: It's a builtin!

**Review by horthbynorthwest — APPROVED** [review]
Approving as this code is ok, but there might be some further tweaks needed to our repo as how we use it might need some work. (i.e. inconsistent search functionality, overlapping possible issues with storages and thumbnailer)

**Pull request merged**